cmd/lk/agent.go

```diff
@@ -522,19 +524,21 @@ func deployAgent(ctx context.Context, cmd *cli.Command) error {
 		return fmt.Errorf("config file [%s] required to update agent", util.Accented(tomlFilename))
 	}
 
-	secrets, err := requireSecrets(ctx, cmd)
-	if err != nil {
-		return err
-	}
-
 	req := &lkproto.DeployAgentRequest{
 		AgentName:   agentConfig.Name,
-		Secrets:     secrets,
 		Replicas:    int32(agentConfig.Replicas),
 		CpuReq:      string(agentConfig.CPU),
 		MaxReplicas: int32(agentConfig.MaxReplicas),
 	}
 
+	secrets, err := requireSecrets(ctx, cmd, false, true)
+	if err != nil {
+		return err
+	}
+	if len(secrets) > 0 {
+		req.Secrets = secrets
+	}
```

Comment:
@real-danm just check my math here, it's totally fine to deploy with no secrets right, because we assume there were valid secrets in the initial creation?

Reply:
yep correct